tests/validate/aie_pl_test/src/pl_controller.cpp

```diff
@@ -0,0 +1,554 @@
+        rtp.selector_lock_id = rtp_node.second.get<uint16_t>("selector_lock_id");
+        rtp.selector_addr = rtp_node.second.get<uint64_t>("selector_address");
+
+        rtp.ping_row = rtp_node.second.get<uint16_t>("ping_buffer_row");
+        rtp.ping_col = rtp_node.second.get<uint16_t>("ping_buffer_column");
+        rtp.ping_lock_id = rtp_node.second.get<uint16_t>("ping_buffer_lock_id");
+        rtp.ping_addr = rtp_node.second.get<uint64_t>("ping_buffer_address");
+
+        rtp.pong_row = rtp_node.second.get<uint16_t>("pong_buffer_row");
+        rtp.pong_col = rtp_node.second.get<uint16_t>("pong_buffer_column");
+        rtp.pong_lock_id = rtp_node.second.get<uint16_t>("pong_buffer_lock_id");
+        rtp.pong_addr = rtp_node.second.get<uint64_t>("pong_buffer_address");
+
+        rtp.is_plrtp = rtp_node.second.get<bool>("is_PL_RTP");
+        rtp.is_input = rtp_node.second.get<bool>("is_input");
+        rtp.is_async = rtp_node.second.get<bool>("is_asynchronous");
+        rtp.is_connected = rtp_node.second.get<bool>("is_connected");
+        rtp.require_lock = rtp_node.second.get<bool>("requires_lock");
+
+        rtps.emplace(std::move(rtp.name), std::move(rtp));
+    }
+}
+
+std::vector<tile_type> plController::get_tiles(const std::string& graph_name) {
+    // ptree aie_meta;
+    // auto data = get_aie_section();
+    // read_aie_metadata(data.first, data.second, aie_meta);
+
+    ptree aie_meta;
+    std::ifstream jsonFile(aie_info_path);
+    if (!jsonFile.good()) throw std::runtime_error("get_tiles():ERROR:No aie info file specified");
+
+    read_json(aie_info_path, aie_meta);
+    std::vector<tile_type> tiles;
+
+    for (auto& graph : aie_meta.get_child("aie_metadata.graphs")) {
+        if (graph.second.get<std::string>("name") != graph_name) continue;
+
+        int count = 0;
+        for (auto& node : graph.second.get_child("core_columns")) {
+            tiles.push_back(tile_type());
+            auto& t = tiles.at(count++);
+            t.col = std::stoul(node.second.data());
+        }
+
+        int num_tiles = count;
+        count = 0;
+        for (auto& node : graph.second.get_child("core_rows")) tiles.at(count++).row = std::stoul(node.second.data());
+        if (count < num_tiles) throw std::runtime_error("core_rows < num_tiles");
+
+        count = 0;
+        for (auto& node : graph.second.get_child("iteration_memory_columns"))
+            tiles.at(count++).itr_mem_col = std::stoul(node.second.data());
+        if (count < num_tiles) throw std::runtime_error("iteration_memory_columns < num_tiles");
+
+        count = 0;
+        for (auto& node : graph.second.get_child("iteration_memory_rows"))
+            tiles.at(count++).itr_mem_row = std::stoul(node.second.data());
+        if (count < num_tiles) throw std::runtime_error("iteration_memory_rows < num_tiles");
+
+        count = 0;
+        for (auto& node : graph.second.get_child("iteration_memory_addresses"))
```

Comment:
It is evident tht this file does not match our coding style.  Please updated it to do so.